include/cppkafka/utils/backoff_committer.h

```diff
@@ -127,25 +135,30 @@ class CPPKAFKA_API BackoffCommitter : public BackoffPerformer {
      */
     Consumer& get_consumer();
 private:
-    // Return true to abort and false to continue committing
-    template <typename T>
-    bool do_commit(const T& object) {
+    // If the ReturnType contains 'true', we abort committing. Otherwise we continue.
+    // The second member of the ReturnType contains the RdKafka error if any.
+    template <typename...Args>
+    bool do_commit(Args&&...args) {
         try {
-            consumer_.commit(object);
-            // If the commit succeeds, we're done
+            consumer_.commit(std::forward<Args>(args)...);
+            // If the commit succeeds, we're done.
             return true;
         }
         catch (const HandleException& ex) {
+            Error error = ex.get_error();
             // If there were actually no offsets to commit, return. Retrying won't solve
             // anything here
-            if (ex.get_error() == RD_KAFKA_RESP_ERR__NO_OFFSET) {
-                return true;
+            if (error == RD_KAFKA_RESP_ERR__NO_OFFSET) {
+                throw ex; //abort
```

Comment:
I think this should probably just return `true`? At least that's what was happening before. I don't really like this error, as it's a non-error error. Committing when there's no offsets is a no-op so I don't really know why it should fail. In my mind this error should be ignored everywhere. Can you think of any case where you'd actually want to be notified that there's nothing to commit?

Reply:
I see your point. Not sure really. The rdkafka documentation states:
 >If no partitions had valid offsets to commit this callback will be called with \p err == RD_KAFKA_RESP_ERR__NO_OFFSET which is not to be considered an error.

So yes I think it's safe to just return true in this case.

Reply:
lol nice. Yeah, I don't really know why this returns a non-error error. It should IMO just return success as the commit itself didn't fail. This basically means every place where you commit you have to check for this special error, which makes it really annoying for users.

But going back to this, yeah, this should probably keep returning true like before.